scipy/io/tests/test_mmio.py

```diff
@@ -111,6 +127,137 @@ def test_random_rect_real(self):
         b = mmread(fn)
         assert_array_almost_equal(a,b)
 
+
+class TestMMIOSparseCSR(TestMMIOArray):
+
+    def setUp(self):
+        self.tmpdir = mkdtemp()
+        self.fn = os.path.join(self.tmpdir, 'testfile.mtx')
+
+    def tearDown(self):
+        shutil.rmtree(self.tmpdir)
+    
+    def test_simple(self):
+        a = [[1,2],[3,4]]
```

Comment:
I don't think it's important for each test to have its own function.   In practice, the single test failure is enough to signal a problem in these routines, and the person fixing the test failure will have to fix all the cases.    Have a look at the matlab io test functions for many examples of multiple tests in the same test function.